File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int n = cost.length;
        int pprev = cost[0], prev = cost[1];
        for(int i = 2; i < n; ++i) {
            int current = cost[i] + Math.min(pprev, prev);
            pprev = prev;
            prev = current;
        }
        return Math.min(pprev, prev);
    }
}

File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        n, pprev, prev = len(cost), cost[0], cost[1]
        for i in range (2, n):
            current = cost[i] + min(pprev, prev)
            pprev, prev = prev, current
        return min(pprev, prev);